### PR TITLE
horizon: 20190716 -> 1.0.0

### PR DESCRIPTION
**Pull request by Luz**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/electronics/horizon-eda/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	version = "20190227";
+
+	src = fetchFromGitHub {
+		owner = "carrotIndustries";
+		repo = "horizon";
+		rev = "810fc8d1f59096ba3e1aac9d4b2187b1010b202a";
+		sha256 = "0qxw3vsbf9cg6b3pp3z1k2fjl8xrnjdr2gcncnr5jzd1mix6yd4y";
+	};
+
+	# Version in specific format, created by: "git log -1 --pretty="format:%h %ci %s"
```
Comment (joachifm) [inline pkgs/applications/science/electronics/horizon-eda/default.nix]: Updating this by hand seems like it'd get annoying after a while. What about just taking part of `src.rev` or does it have to have that format? Or at least use git within the build to produce the thing.

Also note that overriding `patchPhase` prevents users from passing additional patches via `patches`; not necessarily a problem, but something to be aware of.

File: pkgs/applications/science/electronics/horizon-eda/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	patchPhase = ''
+		mkdir .git && touch .git/HEAD && touch .git/index
+		echo "const char *gitversion = \"810fc8d 2019-02-27 00:05:27 +0100 fix some includes\";" > src/gitversion.cpp
+	'';
+
+	buildInputs = [ pkgconfig sqlite libyamlcpp libuuid gnome3.gtkmm epoxy librsvg zeromq cppzmq glm libgit2 curl boost python3 opencascade ];
```
Comment (joachifm) [inline pkgs/applications/science/electronics/horizon-eda/default.nix]: I think `pkgconfig` makes more sense as a native build input

File: pkgs/applications/science/electronics/horizon-eda/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+	nativeBuildInputs = [ wrapGAppsHook ];
+
+	NIX_CFLAGS_COMPILE = "-I${opencascade}/include/oce";
+
+	installPhase = ''
+		mkdir -p $out/bin
```
Comment (joachifm) [inline pkgs/applications/science/electronics/horizon-eda/default.nix]: Golfing: I believe this reduces to `install -Dt $out/bin horizon-{eda,imp}`

**Comment by joachifm** [comment]
@GrahamcOfBorg build horizon-eda

File: pkgs/applications/science/electronics/horizon-eda/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchFromGitHub, pkgconfig, sqlite, libyamlcpp, libuuid, gnome3, epoxy, librsvg, zeromq, cppzmq, glm, libgit2, curl, boost, python3, opencascade, libzip, podofo, wrapGAppsHook }:
+
+stdenv.mkDerivation rec {
+	name = "horizon-eda-${version}";
```
Comment (aanderse) [inline pkgs/applications/science/electronics/horizon-eda/default.nix]: Please use `pname = "horizon-eda";` instead as `name = "${pname}-${version}";` is already implied.

File: pkgs/applications/science/electronics/horizon-eda/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	version = "20190716";
```
Comment (aanderse) [inline pkgs/applications/science/electronics/horizon-eda/default.nix]: > If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23".

File: pkgs/applications/science/electronics/horizon-eda/default.nix [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+		repo = "horizon";
+		rev = "85c171a3dc29d6cab41f5effd230d1769e519b4d";
+		sha256 = "012vi91bxzbvabg6v2ldl916jg24i6pc9lpnniwq3c2pzpvlvndg";
+	};
+
+	patchPhase = ''
+		echo "const char *gitversion = \"${src.rev}\";" > src/gitversion.cpp
+	'';
+
+	buildInputs = [ pkgconfig sqlite libyamlcpp libuuid gnome3.gtkmm epoxy librsvg zeromq cppzmq glm libgit2 curl boost python3 libzip podofo opencascade ];
```
Comment (aanderse) [inline pkgs/applications/science/electronics/horizon-eda/default.nix]: Please see comment by @joachifm about `pkgconfig` belonging to `nativeBuildInputs` instead.

File: pkgs/applications/science/electronics/horizon-eda/default.nix [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+	preFixup = ''
+		wrapProgram "$out/bin/horizon-eda" "''${gappsWrapperArgs[@]}"
+	'';
+
+	enableParallelBuilding = true;
+
+	meta = with stdenv.lib; {
+		description = "A free EDA software to develop printed circuit boards";
+		homepage = https://github.com/carrotIndustries/horizon;
```
Comment (aanderse) [inline pkgs/applications/science/electronics/horizon-eda/default.nix]: Please quote.

File: pkgs/applications/science/electronics/horizon-eda/default.nix [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+	NIX_CFLAGS_COMPILE = "-I${opencascade}/include/oce";
+
+	installPhase = ''
+		mkdir -p $out/bin
+		cp build/horizon-eda $out/bin/
+		cp build/horizon-imp $out/bin/
+	'';
+
+	preFixup = ''
+		wrapProgram "$out/bin/horizon-eda" "''${gappsWrapperArgs[@]}"
```
Comment (aanderse) [inline pkgs/applications/science/electronics/horizon-eda/default.nix]: Is this actually required?

**Review by aanderse — CHANGES_REQUESTED** [review]
I have left a few comments I hope you will find helpful.

File: pkgs/applications/science/electronics/horizon-eda/default.nix [hunk 9]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub, pkgconfig, sqlite, libyamlcpp, libuuid, gnome3, epoxy, librsvg, zeromq, cppzmq, glm, libgit2, curl, boost, python3, opencascade, libzip, podofo, wrapGAppsHook }:
+
+stdenv.mkDerivation rec {
+	pname = "horizon-eda";
+	version = "1.0.0";
+
+	src = fetchFromGitHub {
+		owner = "horizon-eda";
+		repo = "horizon";
+		rev = "3f777bd2501b953981d64cf2054c265ce8d65c02";
```
Comment (aanderse) [inline pkgs/applications/science/electronics/horizon-eda/default.nix]:
```suggestion
		rev = "v${version}";
```

File: pkgs/applications/science/electronics/horizon-eda/default.nix [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+		sha256 = "1d40lfgylwl3srqazyjgw9l3n7i82xrvkjcxvpx7ddyzy1d2aqmr";
+	};
+
+	buildInputs = [ sqlite libyamlcpp libuuid gnome3.gtkmm epoxy librsvg zeromq cppzmq glm libgit2 curl boost python3 libzip podofo opencascade ];
```
Comment (aanderse) [inline pkgs/applications/science/electronics/horizon-eda/default.nix]: I would prefer if you were to sort this alphabetically (easier for me to do a quick comparison against https://docs.horizon-eda.org/en/latest/build-linux.html), but more importantly place each entry on a new line, like so:
```
buildInputs = [
  boost
  curl
  ...
  zeromq
];
```

**Review by aanderse — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/electronics/horizon-eda/default.nix (hunk 3 above)
Comment (carrotIndustries) [reply pkgs/applications/science/electronics/horizon-eda/default.nix]: We now have an install target that also installs icons, desktop entry and such.

**Comment by guserav** [comment]
As of #99964 this is merged so this shouldn't be necessary any longer

**Pull request closed without merging**